cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java

```diff
@@ -84,13 +93,35 @@ public AbstractProgramWorkflowRunner(Program workflowProgram, ProgramOptions wor
   }
 
   /**
+   * Returns the {@link ProgramType} supported by this runner.
+   */
+  protected abstract ProgramType getProgramType();
+
+  /**
+   * Rewrites the given {@link Program} to a {@link Program} that represents the {@link ProgramType} as
+   * returned by {@link #getProgramType()}.
+   */
+  protected abstract Program rewriteProgram(String name, Program program);
```

Comment:
I'm trying to understand what this re-write does. Does it take a program and convert it into a corresponding workflow program? Some more details in the javadoc would be helpful.